apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+window.ReactDOM = require('react-dom');
+window.Radium = require('radium');
+var update = require('react-addons-update');
+
+window.TutorialExplorerManager = function (options) {
+  this.options = options;
+  var self = this;
+
+  const FilterChoice = React.createClass({
+    propTypes: {
+      onUserInput: React.PropTypes.func,
+      groupName: React.PropTypes.string,
+      name: React.PropTypes.string,
+      isCheckedInput: React.PropTypes.bool,
+      value: React.PropTypes.string,
+      selected: React.PropTypes.bool,
+      text: React.PropTypes.string
+    },
+
+    handleChange: function () {
+      this.props.onUserInput(
+        this.props.groupName,
+        this.props.name,
+        this.refs.isCheckedInput.checked
+      );
+    },
+
+    render() {
+      return (
+        <div  style={{userSelect: 'none', WebkitUserSelect: 'none', MozUserSelect: 'none', MsUserSelect: 'none'}}>
+          <label style={{fontFamily: "\"Gotham 4r\", sans-serif", fontSize: 13, paddingBottom: 0, marginBottom: 0, cursor: 'pointer'}}>
+            <input
+              type="checkbox"
+              value={this.props.value}
+              checked={this.props.selected}
+              ref="isCheckedInput"
+              onChange={this.handleChange}
+              style={{marginRight: 5}}
+            />
+            {this.props.text}
+          </label>
+        </div>
+      );
+    }
+  });
+
+  const FilterGroup = React.createClass({
+    propTypes: {
+      name: React.PropTypes.string,
+      text: React.PropTypes.string,
+      filterEntries: React.PropTypes.array,
+      selection: React.PropTypes.array,
+      onUserInput: React.PropTypes.func
+    },
+
+    render() {
+      return (
+        <div style={{paddingTop: 20, paddingRight: 40}}>
+          <div style={{fontFamily: '"Gotham 5r", sans-serif', borderBottom: 'solid grey 1px'}}>
+            {this.props.text}
+          </div>
+          {this.props.filterEntries.map(item => <FilterChoice groupName={this.props.name} name={item.name} text={item.text} selected={this.props.selection && this.props.selection.includes(item.name)} onUserInput={this.props.onUserInput} key={item.name}/>)}
+        </div>
+      );
+    }
+  });
+
+  const FilterSet = React.createClass({
+    propTypes: {
+      filterGroups: React.PropTypes.array,
+      onUserInput: React.PropTypes.func,
+      selection: React.PropTypes.object
+    },
+
+    render() {
+      return (
+        <div>
+          <div className="col-20">
+            <div style={{fontSize: 16}}>
+              Filter By
+            </div>
+            {this.props.filterGroups.map(item => <FilterGroup name={item.name} text={item.text} filterEntries={item.entries} onUserInput={this.props.onUserInput} selection={this.props.selection[item.name]} key={item.name}/>)}
+          </div>
+        </div>
+      );
+    }
+  });
+
+  // For a string of tags, generate a string of their friendly names.
+  function getTagString(prefix, tagString) {
```

Comment:
Perhaps @param tags describing inputs. It looks like we expect tagString to be a string of comma separated strings? Also potentially worth a unit test.